File: app/src/main/java/com/example/rafaelanastacioalves/moby/repository/AppRepository.java
package com.example.rafaelanastacioalves.moby.repository;

import com.example.rafaelanastacioalves.moby.vo.EntityDetails;
import com.example.rafaelanastacioalves.moby.vo.MainEntity;

import java.util.ArrayList;
import java.util.List;

import io.reactivex.Observable;
import io.reactivex.schedulers.Schedulers;

public class AppRepository {

    public static Observable<List<MainEntity>> getMainEntityList(){
        APIClient APIClient = ServiceGenerator.createService(APIClient.class);
        Observable<List<MainEntity>> finalResult = Observable.combineLatest(
                APIClient.getTripPackageList().subscribeOn(Schedulers.io()),
                APIClient.getTripPackageListAdditional().subscribeOn(Schedulers.io()),
                (l1, l2) -> {
                    List<MainEntity> finalList = new ArrayList<>();
                    finalList.addAll(l1);
                    finalList.addAll(l2);
                    return finalList;
                });

    return finalResult;
    }

    public static Observable<EntityDetails> getEntitiDetail(String tripPackageId) {

        APIClient APIClient = ServiceGenerator.createService(APIClient.class);
        return APIClient.getTripPackageDetails(tripPackageId);

    }
}

File: SharedCode/src/commonMain/kotlin/repository/AppRepository.kt
package com.example.rafaelanastacioalves.moby.retrofit

import domain.ServiceLocator
import domain.domain.entities.EntityDetails
import domain.domain.entities.MainEntity
import domain.domain.entities.Resource
import repository.APIClient


object AppRepository {

    suspend fun mainEntity(): Resource<List<MainEntity>> {
        return object : NetworkBoundResource<List<MainEntity>, List<MainEntity>>() {
            override suspend fun makeCall(): List<MainEntity>? {

                var apiClient: APIClient = ServiceLocator.moviesApi
                return apiClient.getTripPackageList()
            }

        }.fromHttpOnly()
    }

    suspend fun mainEntityAdditional(): Resource<List<MainEntity>> {
        return object : NetworkBoundResource<List<MainEntity>, List<MainEntity>>() {
            override suspend fun makeCall(): List<MainEntity>? {

                var apiClient: APIClient = ServiceLocator.moviesApi
                return apiClient.getTripPackageListAdditional()
            }

        }.fromHttpOnly()
    }

    suspend fun entityDetails(requestId: String) : Resource<EntityDetails> {
        return object : NetworkBoundResource<EntityDetails, EntityDetails>(){
            override suspend fun makeCall(): EntityDetails? {
                var apiClient: APIClient = ServiceLocator.moviesApi
                return apiClient.getTripPackageDetails(requestId)
            }
        }.fromHttpOnly()
    }
}